backend/src/apiserver/resource/resource_manager.go

```diff
@@ -340,9 +343,29 @@ func (r *ResourceManager) CreatePipelineAndPipelineVersion(p *model.Pipeline, pv
 	tmpl, err := template.New(pipelineSpecBytes)
 	if err != nil {
 		return nil, nil, util.Wrap(err, "Failed to create a pipeline and a pipeline version due to template creation error")
-	} else if tmpl.IsV2() {
-		tmpl.OverrideV2PipelineName(p.Name, p.Namespace)
 	}
+	// Validate pipeline's name in:
+	// 1. pipeline spec for v2 pipelines and v2-compatible pipeline must comply with MLMD requirements
+	// 2. display name must be non-empty
+	pipelineSpecName := ""
+	if tmpl.IsV2() {
+		pipelineSpecName = tmpl.V2PipelineName()
+		if err := common.ValidatePipelineName(pipelineSpecName); err != nil {
+			return nil, nil, err
+		}
+	}
+	if pv.Name == "" && p.Name == "" {
```

Comment:
Would this ever happen? not required input for v1 API?
Anyway, this contradicts to the comment above: "2. display name must be non-empty"

Reply:
I am probably missing something here. What we are doing here:
1. Validate `spec.pipeline-info.name`
2. If `DisplayName` is empty, replace it with `spec.pipeline-info.name`
3. If both are empty, throw an error.

`DisplayName` can be empty. We can't rely on the validation in UI or SDK. Hence we have the logic to validate it here.